lib/crystalball/rspec/prediction_pruning/examples_pruner.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Crystalball
+  module RSpec
+    module PredictionPruning
+      # A class to prune given world example groups to fit the limit.
+      class ExamplesPruner
+        attr_reader :world, :to
+
+        # @param [RSpec::Core::World] rspec_world RSpec world instance
+        # @param [Integer] to upper bound limit for prediction.
+        def initialize(rspec_world, to:)
+          @world = rspec_world
+          @to = to
+        end
+
+        # @return [Array<String>] set of example and context ids to run
+        def pruned_set
+          self.resulting_set = []
+          self.resulting_size = 0
+
+          world.ordered_example_groups.each { |g| prune_to_limit(g) }
+
+          resulting_set
+        end
+
+        private
+
+        attr_accessor :resulting_set, :resulting_size
+
+        def prune_to_limit(group)
+          return if resulting_size >= to
+
+          group_size = world.example_count([group])
+
+          if resulting_size + group_size > to
+            (group.descendants - [group]).each do |g|
+              prune_to_limit(g)
+            end
+
+            add_examples(group)
+          else
+            add(group, group_size)
+          end
+        end
+
+        def add(group_or_example, group_size)
```

Comment:
This method name is not descriptive. Should be something like `add_group_or_example_to_result`. Also `group_size` can default to `1`, making the addition of a single example or group read better.